frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1479,6 +1490,49 @@ export const dashboardLogic = kea<dashboardLogicType>([
         },
     })),
 
+    subscriptions(({ values, actions }) => ({
+        variables: (variables) => {
+            // try to convert url variables to variables
+            const urlVariables = values.maybeUrlVariables
+            const overrideVariables = []
+
+            for (const [key, value] of Object.entries(urlVariables)) {
+                const variable = variables.find((variable: HogQLVariable) => variable.code_name === key)
+                if (variable) {
+                    overrideVariables.push({ ...variable, value })
```

Comment:
Nit: instead of a `.push` here, can we just call `actions.overrideVariableValue(variable.id, variable.value)` directly to remove this extra `overrideVariables` array and loop below?